Python/pythonrun.c

```diff
@@ -1040,6 +1040,79 @@ print_exception_message(struct exception_print_context *ctx, PyObject *type,
     return 0;
 }
 
+static int
+print_exception_suggestions(struct exception_print_context *ctx,
+                            PyObject *value)
+{
+    PyObject *f = ctx->file;
+    PyObject* suggestions = _Py_Offer_Suggestions(value);
+    if (suggestions) {
+        // Add a trailer ". Did you mean: (...)?"
+        if (PyFile_WriteString(". Did you mean: '", f) < 0) {
+            goto error;
+        }
+        if (PyFile_WriteObject(suggestions, f, Py_PRINT_RAW) < 0) {
+            goto error;
+        }
+        if (PyFile_WriteString("'?", f) < 0) {
+            goto error;
+        }
+        Py_DECREF(suggestions);
+    } else if (PyErr_Occurred()) {
+        PyErr_Clear();
+    }
+    return 0;
+error:
+    Py_XDECREF(suggestions);
+    return -1;
+}
+
+static int
+print_exception_note(struct exception_print_context *ctx,
+                     PyObject *value)
+{
+    PyObject *f = ctx->file;
+    _Py_IDENTIFIER(__note__);
+
+    PyObject *note = _PyObject_GetAttrId(value, &PyId___note__);
+    if (note == NULL) {
+        return -1;
+    }
+    if (!PyUnicode_Check(note)) {
+        Py_DECREF(note);
+        return 0;
+    }
+
+    _Py_static_string(PyId_newline, "\n");
+    PyObject *lines = PyUnicode_Split(
+        note, _PyUnicode_FromId(&PyId_newline), -1);
```

Comment:
Completely OT:

- Silly question: Why do we not use `PyUnicode_InternFromString` instead of `_Py_static_string`+`_PyUnicode_FromId` here?
-  If `_PyUnicode_FromId` fails, `PyUnicode_Split` will default to use whitespace as separator. Is that ok? IMO, we should check that the separator object is not `NULL` before proceeding to `PyUnicode_Split`.

Reply:
Because I don't know what I'm doing :)  (I added this recently, I'll change it).

Reply:
Actually why not PyUnicode_Splitlines?